RxCocoa/Common/DelegateProxyType.swift

```diff
@@ -210,6 +213,14 @@ extension DelegateProxyType {
             proxy.setForwardToDelegate(nil, retainDelegate: retainDelegate)
         }
     }
+    
+    public static func extend<Object: AnyObject>(with factory: @escaping ((Object) -> AnyObject)) {
+        factories.append({ ($0 as? Object).map(factory) })
+    }
+    
+    public static func createProxyForObject(_ object: AnyObject) -> AnyObject {
+        return factories.reversed().reduce(AnyObject?.none) { $0 ?? $1(object) }!
```

Comment:
We could probably name it something more idiomatic, like `createProxy(for object: AnyObject)`